### PR TITLE
Trivialize some securecookie rulesets too

### PR DESCRIPTION
**Pull request by RReverser**
A follow-up to https://github.com/EFForg/https-everywhere/pull/12294.

Figured that we can also safely rewrite rulesets with `securecookie` if list of targets wasn't changed at all - turns out this reveals another 138 rulesets that can be trivialized. cc @Hainish

### CODE REVIEW

**Comment by RReverser** [comment]
@cschanaj @Bisaloo @Hainish Does anyone want to merge this?

File: utils/trivialize-rules/trivialize-rules.js [hunk 1]
```diff
@@ -150,42 +151,48 @@ files.fork().zipAll([ sources.fork(), rules ]).map(([name, source, ruleset]) =>
...
-  info`trivialized`;
-
-  domains = Array.from(new Set(domains));
+  domains = Array.from(domains);
 
   if (domains.slice().sort().join('\n') !== targets.sort().join('\n')) {
+    if (ruleset.securecookie) {
+      return;
+    }
+
```
Comment (Hainish) [inline utils/trivialize-rules/trivialize-rules.js]: I'm not following the logic here.  I see this was removed from the top.  I'm assuming you're using a `Set` now for its property of having unique values.  That would make sense if you were, for instance, lumping all the `target`s into the same `Set` with the `securecookie` hosts and then checking coverage by the `rule` via this exploded `domains`.  But I don't see that happening above.  How did this result in actually checking coverage of the `securecookie`s?

Comment (Hainish) [reply utils/trivialize-rules/trivialize-rules.js]: I can see this now.

This is for when we have a left-wildcard in `target`s, and the `rule` contains multiple subdomains.  In this case, the `target`s have to be expanded to match the newly trivialized `rule`.  But this is complicated if we have a `securecookie`.

So in this more complicated case, we can abort.

I suppose we can we do something like we did with the `rule` tag and explode the regex, seeing if only the expanded `target`s are contained in the `securecookie` regex.  But this is a good iterative step.

Comment (RReverser) [reply utils/trivialize-rules/trivialize-rules.js]: The only difference with previous PR is that the securecookie rule was relaxed a bit. Previously I was bailing out if ruleset contains any securecookie item, but now also walking through cases where exploded regexp doesn't change list of targets, but can be still trivialized - then it's safe to rewrite such ruleset even though it has securecookie.

So yes, it's just another iteration step that is still guaranteed to be backwards-compatible.

Comment (RReverser) [reply utils/trivialize-rules/trivialize-rules.js]: The `Set` changes are unrelated - they are only for better warning messages (previously some hosts were repeated several times in one message).

**Pull request merged**